client/src/queries/tip.ts

```diff
@@ -0,0 +1,33 @@
+import { QueryFunction, queryOptions } from "@tanstack/react-query";
+import * as api from "./fetch/fetchWrapper";
+
+const fetchTip: QueryFunction<
+  UserArtistTip,
+  ["fetchTip", { artistId: string; artistSlug: string; tipId: string }]
+> = ({ queryKey: [_, { artistId, artistSlug, tipId }], signal }) => {
+  return api
+    .get<{
+      result: UserArtistTip;
+    }>(`/v1/artists/${artistId}/${artistSlug}/${tipId}`, {
+      signal,
+    })
+    .then((r) => r.result);
+};
```

Comment:
you don't need both `artistUrl` and `artistId`, the api handles both (because `artistSlug` is unique). 

```suggestion
const fetchTip: QueryFunction<
  UserArtistTip,
  ["fetchTip", { artistId: number; tipId: string }]
> = ({ queryKey: [_, { artistId, tipId }], signal }) => {
  return api
    .get<{
      result: UserArtistTip;
    }>(`/v1/artists/${artistId}/${tipId}`, {
      signal,
    })
    .then((r) => r.result);
};
```

However, I think you'll need to add the endpoint for this in the API.

Reply:
Resolved in [6ac18c4](https://github.com/funmusicplace/mirlo/pull/963/commits/6ac18c4c777ec8d847ee34ffe64e9879560cefb2)